pkg/runtime/reconciler.go

```diff
@@ -85,12 +87,38 @@ func (r *reconciler) BindControllerManager(mgr ctrlrt.Manager) error {
 }
 
 // SecretValueFromReference fetches the value of a Secret given a
-// SecretReference
+// SecretKeyReference.
 func (r *reconciler) SecretValueFromReference(
-	ref *corev1.SecretReference,
-) (string, error) {
-	// TODO(alina-kim): Implement this method :)
-	return "", ackerr.NotImplemented
+	ctx context.Context,
+	ref *acktypes.SecretKeyReference,
+) (*string, error) {
+
+	if ref == nil {
+		return nil, nil
+	}
+
+	namespace := ref.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	secret, err := r.clientSet.CoreV1().Secrets(namespace).Get(ctx, ref.Name, metav1.GetOptions{})
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Instead of the above use of clientset, you can just use the controller-runtime's client, which I think is actually more elegant and shorter:

```go
	nsn := client.ObjectKey{
		Namespace: ref.Namespace,
		Name: ref.Name,
	}
	var secret corev1.Secret
	if err := r.kc.Get(ctx, nsn, &secret); err != nil {
		return nil, err
	}
```

Note that `client.ObjectKey` is simply an alias for apimachinery's `types.NamespacedName`:

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#ObjectKey